### ISSUE TITLE
Query Loop Builder

### ISSUE THREAD

**Issue by rjwdebruin** [beaverbuilder/feature-requests#291]
A query loop builder would be a very useful addition for Beaver Builder to style the post grid the way you want. Too frequently I have to switch the post module to "Custom layout". For example if I want to insert an ACF field in the output.

![Post Layout](https://user-images.githubusercontent.com/81624230/202232017-94d6f94a-4b55-459b-af42-c33217741367.png)

I like the approach that Generate Blocks has chosen, see their [documentation](https://docs.generateblocks.com/article/query-loop-overview/). It could also be an extra function for Beaver Themer.

**Comment by jodzeee** [comment]
Yes! I was just sharing that frustration with PowerPack - there's a bunch of styles to choose from, but if you want to customize them, choosing "custom layout" gives you default code. It'd be nice to be able to start with a template and have multiple options.

**Comment by joshwillett** [comment]
Yes please

**Comment by zackpyle** [comment]
Agreed. I write a lot of custom layouts but know that a lot of people don't have that ability. So this is a very needed feature for those people. I hate Elementor, and I've never used this, but they do have a nice looking loop builder: https://elementor.com/blog/introducing-the-loop-builder/

**Comment by Bobolinos** [comment]
+ a million, Breakdance and Elementor also have this feature

**Comment by dmvdev01** [comment]
I added a similar request that has been merged into this one. This was the original text:

I'd like to create a saved row that I would style as a single post item that becomes part of a new module called post loop builder. This module loads the saved row and repeats it for all the posts. In the saved row itself I connect all fields like the image, post titel, button url etc with the Beaver Themer field connector.

The post loop module shows all these saved rows / saved blocks in a CSS grid or slider or whatever and loads the dynamic values for each block like post title, featured image etc.
We also need a super advanced query with it to filter the (custom) posts.

You don't have to figure out how to create this because it has already been done. If you would just create something like this video below, that would be really awesome!

https://www.youtube.com/watch?v=EM6mfgQQHlA
Other builders like Elementor have this feature too.
It's nice that you can use the BB modules to create a single post item.

PS: The saved row could also be a template if that is more convenient to do, but I hope you get the idea.

**Comment by dmvdev01** [comment]
@codente, good to know that this is on your radar. It would be really nice if you could share your ideas about this in the FB community group for example so we can add other wishes of users. I think the BB should consult the community more, there are lots of good ideas out there.

**Comment by codente** [comment]
We don't have anything concrete at this time as it still has to be researched and figure out the best approach from the development side.

One option we've considered is a new Themer layout type. We're not sure that is the best approach though.  There is also stuff we are working on that might make this easier so first we're going to focus on those things and then we can dig into something like this.

We take and accept feedback from everywhere - we are active in the slack, discord, fb group plus we have this and we also read feedback submitted in tickets.

**Comment by codente** [comment]
This is currently in Progress for Themer.

**Comment by codente** [comment]
This is now in Themer 1.5 which is out now.
